app/controllers/admin/user_groups_controller.rb

```diff
@@ -30,8 +30,16 @@ def edit
     end
 
     def update
-      @group.update(group_params)
-      @group.save
+      users = User.where(id: group_params[:user_ids])
+      users_to_remove = @group.users - users
+      @group.add(users) # add with paper trail
+      @group.remove(users_to_remove) # destroy with paranoia
+
+      if group_params[:name]
+        @group.name = group_params[:name]
+        @group.save!
+      end
```

Comment:
maybe the following, just to catch the situation where we add other fields that should be saved?
(untested)
```suggestion
      @group.update!(group_params.except(:user_ids))
```